hibernate-core/src/main/java/org/hibernate/dialect/function/TemplateRenderer.java

```diff
@@ -61,8 +61,10 @@ public TemplateRenderer(String template) {
 					}
 				}
 
-				paramList.add( Integer.valueOf( index.toString() ) );
-				index.delete( 0, index.length() );
+				if ( index.length() > 0 ) {
+					paramList.add( Integer.valueOf( index.toString() ) );
+				}
+				index.setLength( 0 );
```

Comment:
I think 'setLength()' is a little bit better in terms of both performance and code readability. See https://stackoverflow.com/questions/5192512/how-can-i-clear-or-empty-a-stringbuilder